src/renderer/component/common/form-components/form-field.jsx

```diff
@@ -0,0 +1,61 @@
+// @flow
+import * as React from 'react';
+
+type Props = {
+  name: string,
+  label?: string,
+  render?: () => React.Node,
+  prefix?: string,
+  postfix?: string,
+  error?: string | boolean,
+  helper?: string | React.Node,
+  type?: string,
+  onChange?: any => any,
+  defaultValue?: string | number,
+  placeholder?: string | number,
+};
+
+export class FormField extends React.PureComponent<Props> {
+  render() {
+    const { render, label, prefix, postfix, error, helper, name, type, ...inputProps } = this.props;
+
+    // Allow a type prop to determine the input or more customizability with a render prop
+    let Input;
+    if (type) {
+      Input = () => <input type={type} id={name} {...inputProps} />;
+    } else if (render) {
+      Input = render;
+    }
```

Comment:
@kauffj Added back the type prop back. You are definitely right it makes more sense to keep it. I do think it will be nice to offer extra customization if a consumer wants it. Still need to add something to handle `select` elements.